docs/adr/xcall.md

```diff
@@ -642,6 +651,17 @@ internal function handleRequest(String srcNet, byte[] data) {
     proxyReqs[reqId] = msgReq
 }
 ```
+```
+internal function handleReply(RollbackData rollback, CSMessageRequest reply) {
+    assert rollback.from==reply.to;
+    assert rollback.netTo=reply.from.netId;
+
```

Comment:
We also need to override the protocols of the reply? so that they are the same as the rollback to prevent malicious values in the destination.

Reply:
ya i thought about it too but wasnt sure if we should override or throw error on mismatch.